code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/vnext/alert/AlertPresentable.kt

```diff
@@ -67,4 +69,9 @@ internal class AlertPresentable(
     override fun getPresentation(): Alert {
         return alert
     }
+
+    override fun hasConflicts(visiblePresentations: List<Presentation<*>>): Boolean {
+        // Only show if there are no other alerts or in-app messages visible
+        return visiblePresentations.any { (it is Alert || it is InAppMessage) }
```

Comment:
this determination is actually driven by the `mobileParameters` setting `uiTakeover`. wondering if there's a way to just check for that value, as it also sets us up to automatically handle future "ui takeover" messages as well

Reply:
The `visiblePresentations` does hold all `Presentation` objects which have `PresentationSettings` . So we _can_ can access the `uiTakeover` settings here.
However, looking at the [current code in Core 2.0](https://github.com/search?q=repo%3Aadobe%2Faepsdk-core-android%20uiTakeover&type=code), I see that `uiTakeover` is being used only during determination of whether the message is to be dismissed due to a background tap/back button press  and the decision to show is being done based on presentation delegate + [current state of the tree as seen by `MessagesMonitor`](https://github.com/adobe/aepsdk-core-android/blob/main/code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java#L219). Is this a bug in 2.0 😅  ?

Reply:
Talked with @sbenedicadb on this topic offline. Learned that there have been discussions about allowing multiple UI's to be displayed on the screen simultaneously depending on feasibility (ex: a center aligned modal InAppMessage can co-exist with a banner style InAppMessage when they do not take over/intercept all ui operations - `uiTakeOver = false` ). We discussed that the current implementation of `hadConflicts` merely checking the type of visible presentations will not be sufficient to achieve the behavior.  
We agreed that the current design, however, allows multiple presentations to co-ordinate based on the `PresentationSettings` for each presentation and this can be added later when the feature is needed.